Improve help for `--python-setup-resolver-jobs` to explain its tradeoff

Eric-Arellano opened the pull request:
### Problem

Toolchain encountered several failures in CI when resolving requirements PEXes with an error code of `-9` and no stdout or stderr. An error code of 9 signifies an Out of Memory error (OOM).

We reasoned that this was likely from not configuring `--python-setup-resolver-jobs`. The default value is `None`, which means that we use Pex's default of the # of cores. This is a good default for direct invocations of Pex, but is a bad default for Pants because of Pant's concurrency model. 

Specifically, `--python-setup-resolver-jobs` is configured for each distinct process spawned by Pants. So, the total parallelism is really `= --python-setup-resolver-jobs x --process-execution-local-parallelism`. 

We default `--process-execution-local-parallelism` to `cpu_count x 2`. This means that the default total parallelism for V2 users would be `(cpu_count  x 2) x cpu_count`! For example, with a CPU count of 4 and using the default value for everything, total parallelism would be 32! 

### Solution

Don't change the default because it's too hard to get right and would mess with V1, where the default is good.

But, improve the help message.

Eric-Arellano commented:
> It feels awfully odd to bias an option to v2 when v2 is still not the default. It makes sense for Toolchain and its customers but not Pants (yet).

Fair. If others agree, I'll revert all the changes but keep the `help` message change.

stuhood commented:
The help is helpful, but defaults are challenging for sure:
* it's also the case in v2 that if there is only one `ExecuteProcess` running, that you might want extra concurrency
* remote execution might affect whether this flattens the client machine or not... although the total concurrency might still flatten whatever server is hosting wheels.

Eric-Arellano commented:
I reverted the default change and only kept the `help` change.

@stuhood if I'm not too late, it would be great to include this in 1.26.x.

jsirois commented on src/python/pants/python/python_setup.py:
```diff
@@ -151,7 +151,15 @@ def register_options(cls, register):
-            help="The maximum number of concurrent jobs to resolve wheels with.",
+            help=(
+                "The maximum number of concurrent jobs to resolve wheels with. You may want to "
+                "increase the maximum number of parallel jobs to potentially improve the latency "
```
Wading into more words here in the 2nd sentence brings more confusion in my modelling of a naivev reader. Since the default is num cores but that sentence comes without that context, the naive read is to add a positive number value here bigger than `None`. That won't net a potential speedup until they go to a value greater than num cores and it will in fact likely even hurt performance at that point, even in v1. Adding even more words or axing this bit since it should go without saying esp. in an advanced option seem like the two paths. You know I prefer the latter. The remaining sentences though pull there weight.

Eric-Arellano commented:
Closing because I'm struggling to find wording that addresses review feedback and accommodates both V1 and V2.

Fwit, the new docs now mention this option in the page on CI setup: https://pants.readme.io/docs/using-pants-in-ci.
